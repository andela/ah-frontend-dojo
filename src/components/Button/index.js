// React libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

const Button = (props) => {
  // Destructure props
  const {
    btnClass, btnName, btnEvent, disabled,
  } = props;

  return (

    <button
      className={btnClass}
      type="button"
      onClick={btnEvent}
      disabled={disabled}
    >
      {btnName}
    </button>
  );
};

Button.defaultProps = {
  disabled: false,
};

// Props validation
Button.propTypes = {
  btnName: PropTypes.string.isRequired,
  btnClass: PropTypes.string.isRequired,
  btnEvent: PropTypes.func.isRequired,
  disabled: PropTypes.bool,

};
export default Button;
