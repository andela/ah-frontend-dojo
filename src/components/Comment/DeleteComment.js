// React library
import React, { Fragment } from 'react';

// Third-party libraries
import PropTypes from 'prop-types';

// components
import Button from 'components/Button';

const DeleteComment = ({
  commentId, author, articleSlug, deleteComment, authenticatedUsername
}) => (
  <Fragment>
    {authenticatedUsername === author && (
    <Button
      btnClass="fa fa-trash btn delete"
      btnEvent={(event) => {
        event.preventDefault();
        deleteComment(articleSlug, commentId);
      }}
      btnName=""
      disabled={authenticatedUsername !== author}
    />
    )}
  </Fragment>
);

// PropType validation
DeleteComment.propTypes = {
  authenticatedUsername: PropTypes.string.isRequired,
  articleSlug: PropTypes.string.isRequired,
  commentId: PropTypes.number.isRequired,
  author: PropTypes.string.isRequired,
  deleteComment: PropTypes.func.isRequired,
};

export default DeleteComment;
