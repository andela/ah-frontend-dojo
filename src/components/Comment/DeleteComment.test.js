// React library
import React from 'react';

// Third-party libraries
import { mount } from 'enzyme';

//  Components
import DeleteComment from 'components/Comment/DeleteComment';
import Button from 'components/Button';


describe('Comment Box', () => {
  const deleteFn = jest.fn();
  const props = {
    commentId: 1,
    created_at: '2019-06-15T06:02:00.086733Z',
    author: 'arthur',
    articleSlug: 'my_article_slug',
    deleteComment: deleteFn,
    authenticatedUsername: 'arthur'
  };
  let wrapper = mount(<DeleteComment {...props} />);


  it('should render without crushing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should allow deleting a comment to comment owners only', () => {
    // author same as current user
    expect(wrapper.find(Button).props().disabled).toBe(false);
    const event = {
      target: {
        type: 'button',
        name: 'deleteComment',
      },
      preventDefault: jest.fn(),
    };

    wrapper.find(Button).simulate('click', event);
    expect(deleteFn).toBeCalled();


    // anonymous user
    props.authenticatedUsername = '';
    wrapper = mount(<DeleteComment {...props} />);
    expect(wrapper.text()).toBe('');

    // current user different from comment author
    props.authenticatedUsername = 'Bison';
    props.author = 'Zack';
    wrapper = mount(<DeleteComment {...props} />);
    expect(wrapper.text()).toBe('');
  });
});
