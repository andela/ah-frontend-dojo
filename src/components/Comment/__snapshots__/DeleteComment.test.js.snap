// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Box should render without crushing 1`] = `
<DeleteComment
  articleSlug="my_article_slug"
  authenticatedUsername="arthur"
  author="arthur"
  commentId={1}
  created_at="2019-06-15T06:02:00.086733Z"
  deleteComment={[MockFunction]}
>
  <Button
    btnClass="fa fa-trash btn delete"
    btnEvent={[Function]}
    btnName=""
    disabled={false}
  >
    <button
      className="fa fa-trash btn delete"
      disabled={false}
      onClick={[Function]}
      type="button"
    />
  </Button>
</DeleteComment>
`;
